tracks/python/exercises/robot-name/mentoring.md

```diff
@@ -20,19 +20,16 @@ class Robot(object):
     names = set()
 
     def __init__(self):
-        self.set_name()
-
-    def reset(self):
-        self.set_name()
-
-    def set_name(self):
         while True:
             self.name = ''.join(
                 random.choices(string.ascii_uppercase, 2) + random.choices(string.digits, 3)
             )
             if self.name not in Robot.names:
                 Robot.names.add(self.name)
                 break
+
+    def reset(self):
+        self.__init__()
```

Comment:
To quote from my Robot notes,

> In general, dunder-methods (double-underscore) should not be directly called when it can be helped. They are for internal use. Prefer instead to have `__init__` call -reset()`.